packages/inter-protocol/src/psm/psm.js

```diff
@@ -243,47 +247,73 @@ export const start = async (zcf, privateArgs, baggage) => {
     E(anchorBrand).getAmountShape(),
     E(stableBrand).getAmountShape(),
   ]);
-  const makeWantStableInvitation = () =>
-    zcf.makeInvitation(
-      wantStableHook,
-      'wantStable',
-      undefined,
-      M.split({ give: { In: anchorAmountShape } }),
-    );
-  const makeGiveStableInvitation = () =>
-    zcf.makeInvitation(
-      giveStableHook,
-      'giveStable',
-      undefined,
-      M.split({ give: { In: stableAmountShape } }),
-    );
 
-  const publicFacet = Far('Parity Stability Module', {
-    getMetrics: () => metricsSubscriber,
-    getPoolBalance: () => anchorPool.getAmountAllocated('Anchor', anchorBrand),
-    makeWantStableInvitation,
-    makeGiveStableInvitation,
+  const PSMI = M.interface('PSM', {
+    getMetrics: M.call().returns(M.remotable('MetricsSubscriber')),
+    getPoolBalance: M.call().returns(anchorAmountShape),
+    makeWantStableInvitation: M.call().returns(M.promise()),
+    makeGiveStableInvitation: M.call().returns(M.promise()),
+    ...extendPublicAPI,
   });
 
-  const getRewardAllocation = () => feePool.getCurrentAllocation();
+  const publicFacet = vivifyFarInstance(
+    baggage,
+    'Parity Stability Module',
+    PSMI,
+    {
+      getMetrics() {
+        return metricsSubscriber;
+      },
+      getPoolBalance() {
+        return anchorPool.getAmountAllocated('Anchor', anchorBrand);
+      },
+      makeWantStableInvitation() {
+        return zcf.makeInvitation(
+          wantStableHook,
+          'wantStable',
+          undefined,
+          M.split({
+            give: { In: anchorAmountShape },
+            want: M.or({ Out: stableAmountShape }, {}),
+          }),
+        );
+      },
+      makeGiveStableInvitation() {
+        return zcf.makeInvitation(
+          giveStableHook,
+          'giveStable',
+          undefined,
+          M.split({
+            give: { In: stableAmountShape },
+            want: M.or({ Out: anchorAmountShape }, {}),
+          }),
+        );
+      },
+      ...extendPublicFacet,
+    },
+  );
+
+  // TODO why does this operation return an object with a single operation?
```

Comment:
are you suggesting that, https://github.com/Agoric/agoric-sdk/blob/f338d2e09087dce71df148375e990256f4301bff/packages/inter-protocol/src/collectFees.js#L29

instead be simply,
```js
return makeCollectFeesInvitation;
```

👍  to that, especially since the function is named `makeMakeCollectFeesInvitation`.

Reply:
I *think* that it's nested because of some restriction in some usage where you can't return a function...  But I'll try fixing that if it's easy.

Reply:
If so then I'd suggest the outer maker is returning an `InvitationMakers` type (the thing a continuing invitation returns). Then I'd suggest the outer be renamed to `makeCollectFeesInvitationMakers`. It's awkward because there's just but "InvitationMakers" is the object type.

Alternately, we could come up with a more concise term for "object whose properties are all invitation makers".

Reply:
InvitationMakersKit

Reply:
That is certainly not more concise :-P 

More importantly, doesn't FooKit mean "foo and related"? (à la IssuerKit) This is not returning any more than invitation makers.

Reply:
I'd say *Kit means a set of related things returned as the values of named properties in a copyRecord. I'd say *FooKit says they are related to Foo, whatever that is. For an InvitationMakersKit, I would not be surprised to find a copyRecord of InvitationMakers.

That was off the top of my head. If we have a definition somewhere, I'd be curious how it differs.

Reply:
Buyilding types into names is generally a bad practice (bad hungarian). Instead, what's the purpose or intent?

Reply:
Purpose is to get additional invitations from the result of the settled one. `continuingInvitations`?

Reply:
I like that.